### PR TITLE
ci: add comments for vercel-deploy-preview

### PR DESCRIPTION
**Pull request by pratapaprasanna**
**/build-deploy-preview env=release**

Screenshot 2023-03-14 at 9 49 44 AM

### CODE REVIEW

File: .github/workflows/on-demand-build-docker-image-deploy-preview.yml [hunk 1]
```diff
@@ -79,12 +87,23 @@ jobs:
...
+           issue-number: ${{ github.event.client_payload.pull_request.number }}
+           body: |
+             Deploy-Preview-URL: ${{ steps.set-dpurl.outputs.dpurl }}
+
 
   push-image:
     needs: [client-build, rts-build, server-build]
     runs-on: ubuntu-latest
-    if: success() && github.event.client_payload.slash_command.args.named.env != 'release'
+    if: success()
```
Comment (sharat87) [inline .github/workflows/on-demand-build-docker-image-deploy-preview.yml:106]: Do we want to push the image for this case now? Deployment happens on Vercel if `env == release`, no?

File: .github/workflows/on-demand-build-docker-image-deploy-preview.yml [hunk 2]
```diff
@@ -22,8 +22,10 @@ jobs:
             PR: ${{ github.event.client_payload.pull_request.number }}.
 
   server-build:
-    if: github.event.client_payload.slash_command.args.named.env != 'release'
     name: server-build
+    if: |
+      github.event.client_payload.slash_command.args.named.env != 'release' &&
+      github.event.client_payload.slash_command.args.named.env != 'prod'
```
Comment (sharat87) [inline .github/workflows/on-demand-build-docker-image-deploy-preview.yml:28]: Are we supporting `env=prod` here?

File: scripts/deploy_preview.sh [hunk 3]
```diff
@@ -35,6 +35,9 @@ export DOMAINNAME=ce-"$PULL_REQUEST_NUMBER".dp.appsmith.com
 export HELMCHART="appsmith"
 export HELMCHART_URL="http://helm.appsmith.com"
 export HELMCHART_VERSION="2.0.0"
+export DB_USERNAME=$DB_USERNAME
+export DB_PASSWORD=$DB_PASSWORD
+export DB_URL=$DB_URL
```
Comment (sharat87) [inline scripts/deploy_preview.sh:40]: I don't think this is needed. None of the commands need these env variables, right? We just need them for interpolation in arguments. That'll work without this too.

```suggestion
```

File: scripts/deploy_preview.sh [hunk 4]
```diff
@@ -64,4 +67,5 @@ helm upgrade -i $CHARTNAME appsmith/appsmith -n $NAMESPACE \
   --set "ingress.annotations.service\.beta\.kubernetes\.io/aws-load-balancer-ssl-cert=$AWS_RELEASE_CERT" \
   --set "ingress.hosts[0].host=$DOMAINNAME, ingress.hosts[0].paths[0].path=/, ingress.hosts[0].paths[0].pathType=Prefix" \
   --set ingress.className="nginx" --set image.pullPolicy="Always" --set autoupdate.enabled="true" --set persistence.size=4Gi \
+  --set applicationConfig.APPSMITH_MONGODB_URI="mongodb+srv://$DB_USERNAME:$DB_PASSWORD@$DB_URL/$DBNAME?retryWrites=true&minPoolSize=1&maxPoolSize=10&maxIdleTimeMS=900000&authSource=admin" \
```
Comment (sharat87) [inline scripts/deploy_preview.sh:70]: This is using `DBNAME` env variable, but in the workflow, we aren't setting it. Instead, we are setting `DB_URL`. Is that okay?

File: scripts/deploy_preview.sh (hunk 3 above)
Comment (pratapaprasanna) [reply scripts/deploy_preview.sh:40]: main intention is keep track of variables and nothing more

File: scripts/deploy_preview.sh (hunk 4 above)
Comment (pratapaprasanna) [reply scripts/deploy_preview.sh:70]: My bad

File: scripts/deploy_preview.sh [hunk 5]
```diff
@@ -31,6 +31,7 @@ export AWS_SESSION_TOKEN=$(echo $sts_output | jq -r '.Credentials''.SessionToken
 export NAMESPACE=ce"$PULL_REQUEST_NUMBER"
 export CHARTNAME=ce"$PULL_REQUEST_NUMBER"
 export SECRET=ce"$PULL_REQUEST_NUMBER"
+export DBNAME =ce"$PULL_REQUEST_NUMBER"
```
Comment (sharat87) [inline scripts/deploy_preview.sh:34]: Syntax error. 🙂

```suggestion
export DBNAME=ce"$PULL_REQUEST_NUMBER"
```

**Review by sharat87 — CHANGES_REQUESTED** [review]

**Review by sharat87 — APPROVED** [review]

**Pull request merged**